torch/csrc/jit/runtime/graph_executor.cpp

```diff
@@ -216,6 +216,7 @@ struct DifferentiableGraphBackward : public autograd::Node {
         input_instructions_(input_size) {}
 
   variable_list apply(variable_list&& inputs) override {
+    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
It is probably not re-entrant, but I can't say for sure, because of how self-recursive autograd can be. It certainly can be used from multiple threads and this would essentially block all task parallelism. Why does the lock need to be held?

Reply:
synced offline, GraphDifferentiableBackward is creating local stack and every other operations are read only, so we don't need to lock it. removing.

Reply:
You still want it if there are saved Variables? (like we do in for the codegened functions)

Reply:
yeah but I think GraphDifferentiableBackward does not in flight release variables as other autograd functions, so we are just purely unpacking saved variables without deleting it.

Reply:
It looks like it does not implement "release_variables" at all. So that is fine I guess.
I guess as long as it is thread safe (cc @zdevito is it?), we can remove the lock.